### PR TITLE
monado: 21.0.0 -> unstable-2022-05-28

### PR DESCRIPTION
**Pull request by LunNova**
Upstream hasn't had a version bump in 1.5 years, lots of progress since then.

~~Following [version normalization RFC](https://github.com/AndersonTorres/rfcs/blob/my-rfc/rfcs/0107-version-normalization.md).~~

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/monado/default.nix [hunk 1]
```diff
@@ -46,25 +46,16 @@
...
 
-  patches = [
-    # https://github.com/NixOS/nixpkgs/issues/137245
-    # Fix warning after Vulkan 1.2.174 VK_NULL_HANDLE change
-    (fetchpatch {
-      url = "https://gitlab.freedesktop.org/monado/monado/-/commit/c47775a95d8e139a2f234063793eb6726f830510.patch";
-      sha256 = "093ymvi9ifpk4vyjcwhhci9cnscxwbv5f80xdbppcqa0j92nmkmp";
-    })
-  ];
-
```
Comment (gador) [inline pkgs/applications/graphics/monado/default.nix:67]: Since removing the patches, `fetchPatch` can be removed from above

**Review by gador — CHANGES_REQUESTED** [review]
LGTM otherwise

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 173907` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>monado</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
monado:
<details>

got build log for '/nix/store/kk211xnr7cvidw8h7phb3nl3c2d9hsnb-monado-21.0.0+date=2022-05-21' from 'daemon'
When evaluating attribute ‘monado’:
warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/applications/graphics/monado/default.nix:4:3:

```
  |
4 | , fetchpatch
  |   ^
```
</details>
</details>

File: pkgs/applications/graphics/monado/default.nix (hunk 1 above)
Comment (LunNova) [reply pkgs/applications/graphics/monado/default.nix:67]: Done

File: pkgs/applications/graphics/monado/default.nix [hunk 2]
```diff
@@ -46,25 +45,16 @@
 
 stdenv.mkDerivation rec {
   pname = "monado";
-  version = "21.0.0";
+  version = "21.0.0+date=2022-05-21";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/monado/default.nix:48]:
```suggestion
  version = "unstable-2022-05-21";
```

File: pkgs/applications/graphics/monado/default.nix [hunk 3]
```diff
@@ -46,25 +45,16 @@
...
 
   src = fetchFromGitLab {
     domain = "gitlab.freedesktop.org";
     owner = pname;
     repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/monado/default.nix:53]:
```suggestion
    owner = "monado";
    repo = "monado";
```
makes it easier to get to the source url.

File: pkgs/applications/graphics/monado/default.nix [hunk 4]
```diff
@@ -130,5 +120,8 @@ stdenv.mkDerivation rec {
     license = licenses.boost;
     maintainers = with maintainers; [ expipiplus1 prusnak ];
     platforms = platforms.linux;
+    # Seemed like the most obvious choice, package also includes
+    # monado-ctl, monado-gui, monado-service
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/monado/default.nix:124]:
```suggestion
```

**Comment by expipiplus1** [comment]
@LunNova LGTM, happy for you (or anyone to merge), but could you please squash the commits, thanks :)

**Comment by LunNova** [comment]
Rebased, squashed, and bumped to `unstable-2022-05-28`
